pkg/provider/kubernetes/provider.go

```diff
@@ -127,15 +130,25 @@ func (p *Provider) newClient(ctx context.Context, store esv1beta1.GenericStore,
 		return nil, err
 	}
 
-	config := &rest.Config{
-		Host:        client.store.Server.URL,
-		BearerToken: string(client.BearerToken),
-		TLSClientConfig: rest.TLSClientConfig{
-			Insecure: false,
-			CertData: client.Certificate,
-			KeyData:  client.Key,
-			CAData:   client.CA,
-		},
+	var config *rest.Config
+	if len(client.KubeConfig) > 0 {
+		var err error
+		if config, err = clcmd.BuildConfigFromKubeconfigGetter("", func() (*clcmdapi.Config, error) {
+			return clcmd.Load(client.KubeConfig)
+		}); err != nil {
+			return nil, fmt.Errorf("failed to create kube config - %w", err)
+		}
+	} else {
+		config = &rest.Config{
+			Host:        client.store.Server.URL,
+			BearerToken: string(client.BearerToken),
+			TLSClientConfig: rest.TLSClientConfig{
+				Insecure: false,
+				CertData: client.Certificate,
+				KeyData:  client.Key,
+				CAData:   client.CA,
+			},
+		}
```

Comment:
I find it hard to reason about what is happening here (that's before your PR): In `setAuth()` we're modifying a bunch of fields on the `Client{}` struct, then we construct a `*rest.Config` from those fields and pass that along.
It's not immediately clear where those fields (`client.key`, `client.CA` etc.) are set, and what value they might have and why.

Now, with your change this gets more complex, reading the Kubeconfig in `setAuth()`, storing it in `client.KubeConfig`, passing it into `BuildConfigFromKubeconfigGetter()` which produces the `*rest.Config`. 

The abstraction/design was flawed before your PR, could you please refactor that so it is easier to reason about? This may be an opportunity to eliminate the fields on the client alltogether, that would be great :crossed_fingers: 

Do you have some time to look into that? If not i can try and work something out.

Reply:
Not sure if I got you right, have removed the additional fields on the `Client` struct und directly constructed the `*rest.Config` there. Also adjusted the `auth_test.go` to work with the new structure. Hope this goes in the right direction :-D

Reply:
yeah, that's what i have imagined! Well done!
CI shows lint/test issues, can you take a look at them?

Reply:
ah yes there were some side effects in the provider tests I have not seen